sparse/unit_test/Test_Sparse_spmv.hpp

```diff
@@ -1112,6 +1125,112 @@ void test_github_issue_101() {
   }
 }
 
+template <class scalar_t, class lno_t, class size_type, class layout_t,
+          class DeviceType>
+void test_spmv_all_interfaces_light() {
+  // Using a small matrix, run through the various SpMV interfaces and
+  // make sure they produce the correct results.
+  using execution_space = typename DeviceType::execution_space;
+  using mag_t           = typename Kokkos::ArithTraits<scalar_t>::mag_type;
+  using crsMat_t = typename KokkosSparse::CrsMatrix<scalar_t, lno_t, DeviceType,
+                                                    void, size_type>;
+  Kokkos::Random_XorShift64_Pool<execution_space> rand_pool(13718);
+  const lno_t m      = 111;
+  const lno_t n      = 99;
+  const mag_t maxVal = 10.0;
+  const mag_t eps    = 10.0 * Kokkos::ArithTraits<mag_t>::eps();
+  size_type nnz      = 600;
+  crsMat_t A         = KokkosSparse::Impl::kk_generate_sparse_matrix<crsMat_t>(
+      m, n, nnz, 2, lno_t(n * 0.7));
+  // note: A's values are in range [0, 50)
+  const mag_t maxError = (nnz / m) * 50.0 * maxVal;
+  using multivector_t  = Kokkos::View<scalar_t **, layout_t, DeviceType>;
+  using vector_t       = Kokkos::View<scalar_t *, layout_t, DeviceType>;
+  using range1D_t      = Kokkos::RangePolicy<execution_space>;
+  using range2D_t = Kokkos::MDRangePolicy<execution_space, Kokkos::Rank<2>>;
+  multivector_t x_mv("x_mv", n, 3);
+  vector_t x("x", n);
+  // Randomize x (it won't be modified after that)
+  Kokkos::fill_random(x_mv, rand_pool, randomUpperBound<scalar_t>(maxVal));
+  Kokkos::fill_random(x, rand_pool, randomUpperBound<scalar_t>(maxVal));
+  multivector_t y_mv("y_mv", m, 3);
+  vector_t y("y", m);
+  // Compute the correct y = Ax once
+  multivector_t ygold_mv("ygold_mv", m, 3);
+  vector_t ygold("ygold", m);
+  for (lno_t i = 0; i < 3; i++)
+    Test::sequential_spmv(A, Kokkos::subview(x_mv, Kokkos::ALL(), i),
+                          Kokkos::subview(ygold_mv, Kokkos::ALL(), i), 1.0,
+                          0.0);
+  Test::sequential_spmv(A, x, ygold, 1.0, 0.0);
+  auto clear_y = [&]() { Kokkos::deep_copy(y_mv, scalar_t(0)); };
+  auto verify  = [&]() {
+    int num_errors = 0;
+    Kokkos::parallel_reduce(
+        "KokkosSparse::Test::spmv", range1D_t(0, m),
+        Test::fSPMV<vector_t, vector_t>(ygold, y, eps, maxError), num_errors);
+    EXPECT_EQ(num_errors, 0);
+  };
+  auto verify_mv = [&]() {
+    int num_errors = 0;
+    Kokkos::parallel_reduce("KokkosSparse::Test::spmv",
+                            range2D_t({0, 0}, {m, 3}),
+                            Test::fSPMV<multivector_t, multivector_t>(
+                                ygold_mv, y_mv, eps, maxError),
+                            num_errors);
+    EXPECT_EQ(num_errors, 0);
+  };
+  // Now run through the interfaces and check results each time
+  execution_space exec;
```

Comment:
To ensure no issues with running spmv concurrently on multiple streams, should this be partitioned into > 1 spaces via `Kokkos::Experimental::partition_space`?

Reply:
Done - it now partitions the default space into 2 and runs on the second one